### DISCUSSION TITLE
WithMountedCache uses the engine's default `user:group` combination permissions

### DISCUSSION THREAD

**Discussion by marcosnils** [dagger/dagger#12176] [maintainers]
# Problem

It has been raised by a user in Discord (https://discord.com/channels/707636530424053791/1316771968762253312) that `WithMountedCache` seems to be honoring the engine's default `user:group` permissions. This causes issues as it breaks the pipeline portability depending on how the the engine is provisioned

# Solution

How about if we enforce `root:root` for all `WithCacheVolume` calls **unless** the `Owner` field is set in the call? 

# Steps to reproduce

1. run the engine with a custom `user:group` option. i.e:

`docker run -u root:1001  --rm --privileged -v /var/lib/dagger --name dagger-engine-v0.15.0 registry.dagger.io/engine:v0.15.0`

2. call the pipeline below.

```go
// Returns a container that echoes whatever string argument is provided
func (m *Lala) Test() *dagger.Container {
        return dag.Container().From("docker.io/library/rust:1.81.0@sha256:7b7f7ae5e49819e708369d49925360bde2af4f1962842e75a14af17342f08262").
                WithMountedCache("/cache", dag.CacheVolume("foo")).
                WithExec([]string{"ls", "-la", "/cache"})
}
```

You'll see the default permissions returned by the pipeline above will be effectively `root:1001`

cc @jedevc @sipsma

**Comment by marcosnils** [comment]
bumping this one since another user tripped with it today https://discord.com/channels/707636530424053791/1366222120173965312/1366814159324582019. 

 This also happens to all With* methods that create files or directories in the underlying containers.

**Comment by shykes** [comment]
> How about if we enforce root:root for all WithCacheVolume calls unless the Owner field is set in the call?

@sipsma @alexcb @tiborvass do you all agree with this? If so, I'll open an issue to track. (IMO not actionable as is unless we have consensus on what to do).
